### PR TITLE
more validated tests to fix #344

### LINKED ISSUE #344: Better test coverage for Validated
working on this since I left these out in the first pass

### PR DESCRIPTION
**Pull request by woparry**

### CODE REVIEW

**Comment by non** [comment]
:+1: thanks!

File: tests/src/test/scala/cats/tests/ValidatedTests.scala [hunk 1]
```diff
@@ -16,6 +18,22 @@ class ValidatedTests extends CatsSuite {
   checkAll("Validated[String, Int] with Option", TraverseTests[Validated[String,?]].traverse[Int, Int, Int, Int, Option, Option])
   checkAll("Traverse[Validated[String,?]]", SerializableTests.serializable(Traverse[Validated[String,?]]))
 
+  implicit val arbitraryValidated: Arbitrary[Validated[String, Int]] = Arbitrary {
```
Comment (rossabaker) [inline tests/src/test/scala/cats/tests/ValidatedTests.scala]: [Same question](https://github.com/non/cats/pull/342#discussion_r32244560) I had with ior.

File: tests/src/test/scala/cats/tests/ValidatedTests.scala [hunk 2]
```diff
@@ -16,6 +18,22 @@ class ValidatedTests extends CatsSuite {
...
+      validated <- if (valid) arbitrary[Int].map(Valid(_))
+                   else arbitrary[String].map(Invalid(_))
+    } yield validated
+  }
+
+  implicit val arbitraryTryInt: Arbitrary[Try[Int]] = Arbitrary {
+    for {
+      success <- arbitrary[Boolean]
+      t <- if (success) arbitrary[Int].map(Success(_))
+      else Gen.const(Failure(new Throwable {}))
```
Comment (rossabaker) [inline tests/src/test/scala/cats/tests/ValidatedTests.scala]: Scalacheck comes with an `Arbitrary[Throwable]`, in case you want to be even more arbitrary.

**Comment by woparry** [comment]
Thanks for the feedback!
I didn't notice the arbitrary for validated because it was buried way down in discipline.

I wonder if these should be elsewhere since they're useful for end-users even if they're not using discipline (because they trust us to do the law-checking and are just writing their own unit tests).

That's a separate issue though.

**Comment by non** [comment]
@woparry Yeah, I've started thinking that having a standard way to publish `Gen[_]` and `Arbitrary[_]` instances in addition to laws is important. It's not something I've seen a lot of libraries do well, yet.

**Comment by rossabaker** [comment]
:+1:

**Pull request merged**